### ISSUE TITLE
Loses stream when passed to gulpif

### ISSUE THREAD

**Issue by barneycarroll** [craigjennings11/gulp-uglifyjs#7]
Seems to abort the task *and* abort gulp execution when passed to [gulp-if](https://www.npmjs.org/package/gulp-if). I assume this is something to do with the piping API, but I can't work out how or why it fails.

For example, [this recipe](https://github.com/gulpjs/gulp/blob/master/docs/recipes/pass-params-from-cli.md) won't work:

```
// npm install gulp yargs gulp-if gulp-uglify
var args   = require('yargs').argv;
var gulp   = require('gulp');
var gulpif = require('gulp-if');
var uglify = require('gulp-uglify');

var isProduction = args.type === 'production';

gulp.task('scripts', function() {
  return gulp.src('**/*.js')
    .pipe(gulpif(isProduction, uglify())) // only minify if production
    .pipe(gulp.dest('dist'));
});
```

This gives me the following output:

```
[gulp] Starting 'assets'...
[gulp] gulp-uglifyjs - No files given; aborting minification
```
…Before exiting to the command line. This was all fine before introducing `gulpif` — maybe I should pester them?

**Comment by ilwebdifabio** [comment]
The same of @AgentME :(

**Comment by craigjennings11** [comment]
I believe this problem was fixed via 16b9b51ff3053af8fe2cbb33b761a7dff4e93bad. Closing unless someone reports otherwise.
